python/python/bystro/ancestry/train.py

```diff
@@ -0,0 +1,480 @@
+"""Training code for Global Ancestry Model.
+
+This script takes a vcf of preprocessed variants (see preprocess_vcfs.sh) and generates:
+
+1.  A list of variants to be used for inference.
+
+2.  PCA loadings mapping the list of variants in (1) to PC space.
+
+3.  Classifiers mapping PC space to the 26 HapMap populations as well as 5 continent-level
+superpopulations.
+"""
+import dataclasses
+import logging
+import random
+import sys
+from collections import Counter
+from pathlib import Path
+from typing import Any, Literal, TypeVar, get_args
+
+import allel
+import numpy as np
+import pandas as pd
+import skops
+import tqdm
+from sklearn.decomposition import PCA
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import accuracy_score
+from sklearn.model_selection import train_test_split
+
+from bystro.ancestry.asserts import assert_equals, assert_true
+from bystro.ancestry.train_utils import get_variant_ids_from_callset
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+DATA_ROOT_DIR = Path.home() / "emory/human_ancestry_project/data/1000_genome_VCF"
```

Comment:
In the final version, are you planning on specifying an acceptable project structure, or else config file that you read these paths / parameters from?

Reply:
yeah this is definitely not the way we want to do this in prod.  Given that this is a training job we'll probably end up passing it in as a command line arg.